src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/service/ReferralService.kt

```diff
@@ -287,7 +287,7 @@ class ReferralService(
   }
 
   private fun generateReferenceNumber(referral: Referral): String? {
-    val category = referral.intervention.dynamicFrameworkContract.serviceCategory.name
+    val category = referral.intervention.dynamicFrameworkContract.contractType.serviceCategories.elementAt(0).name
```

Comment:
sorry for comment late in the day - i think we should use contract type to generate the refernce number

Reply:
we can merge this and fix this after